app/tools/android/customize.py

```diff
@@ -331,12 +331,18 @@ def CustomizeJava(app_info, app_url, app_local_path, keep_screen_on):
   dest_activity = os.path.join(app_pkg_dir, name + 'Activity.java')
   ReplaceString(dest_activity, 'org.xwalk.app.template', package)
   ReplaceString(dest_activity, 'AppTemplate', name)
-  manifest_file = os.path.join(app_dir, 'assets', 'www', 'manifest.json')
-  if os.path.isfile(manifest_file):
-    ReplaceString(
-        dest_activity,
-        'loadAppFromUrl("file:///android_asset/www/index.html")',
-        'loadAppFromManifest("file:///android_asset/www/manifest.json")')
+  if manifest:
+    manifest_name = manifest.input_path.split(os.path.sep)[-1]
+    manifest_file = os.path.join(app_dir, 'assets', 'www', manifest_name)
+    if os.path.isfile(manifest_file):
+      manifest_in_asset = 'file:///android_asset/www/' + manifest_name
```

Comment:
Nonetheless, have you checked if the manifest ends up in `www/` if it is in a sub-directory (ie. does `--manifest=foo/bar/my_manifest.json` cause the file to be in `www/my_manifest.json` or `www/foo/bar/my_manifest.json`)?